src/shared/discover-image.c

```diff
@@ -754,29 +753,44 @@ int image_find(RuntimeScope scope,
         if (!names)
                 return -ENOMEM;
 
+        _cleanup_close_ int root_fd = AT_FDCWD;
```

Comment:
Please name this `rfd` which is the naming we use for root fds elsewhere as well